packages/backend/src/tokens/request.ts

```diff
@@ -299,8 +376,13 @@ ${error.getFullMessage()}`,
       if (errors) {
         throw errors[0];
       }
-      // use `await` to force this try/catch handle the signedIn invocation
-      return signedIn(authenticateContext, data, undefined, sessionTokenInHeader!);
+      const signedInRequestState = signedIn(authenticateContext, data, undefined, sessionTokenInHeader!);
+      // Org sync if necessary
+      const handshakeRequestState = handleMaybeOrganizationSyncHandshake(
```

Comment:
AFAIK, we do not currently support handshake for token in header, so I think this should be removed.

Token in header implies that the request is a fetch() rather than a navigation (which would rely on cookies).

The issue is that a fetch request will not normally come up with `credentials: true`, and then redirecting to handshake fails because the __client cookie is not forwarded along to FAPI.

Rather than support this and try to educate on credentials: true, we believe the proper the solution is to include the appropriate token in the fetch(), and I think that logic holds for org sync.

Reply:
👍  Understood. To paraphrase and check my own understanding:

Handshake isn't supported on fetch requests, because a non-browser fetch-er won't automatically include the `__client` cookie, so the handshake response wouldn't be useful. We are currently never handshaking when the auth is defined via headers, and it doesn't make sense org sync either.

I'll pull the handshake detection out of this pathway.

Reply:
One other interesting note: while pairing, @LauraBeatris , @wobsoriano  and I figured out that we will return a handshake response when the auth header is present IF the `sec-fetch-dest: document` header is applied. 

I think that's fine, because no fetch client would naturally include that, but it's an interesting hole in the "we don't redirect if there's header auth" assumption.

Reply:
Can you clarify the hole? `sec-fetch-dest: document` implies it's a page navigation, and the `Authorization` header cannot be set on page navigation (with small exception of Basic auth, but that's not relevant to clerk's header auth that relies on Bearer keys).

Reply:
Just a hole in my own assumptions about when we would and wouldn't handshake. I was assuming we were using the auth method (header vs cookie) to determine if we handshake, but in fact it's the `sec-fetch-dest` and `accept` headers that make the ultimate decision: https://github.com/clerk/javascript/blob/19d3808d4672234944226d6709ec51214e8d6e1d/packages/backend/src/tokens/request.ts#L65-L71

But you're right - the auth header and sec-fetch-dest:document should be mutually exclusive, so the distinction doesn't matter in practice.

Reply:
In any case, fixed: 889097c2eba9af33778ab5b87cd0f3ea9e81b93d